client/src/components/Weather/index.js

```diff
@@ -0,0 +1,134 @@
+    const [celsius, setCelsius] = useState(undefined);
+    const [temp_max, setMaxTemp] = useState(null);
+    const [temp_min, setMinTemp] = useState(null);
+    const [description, setDescription] = useState("");
+   
+    /* the get_WeatherIcon functions set the weather icon based on the weather condition codes
+       see https://openweathermap.org/weather-conditions for more details
+       cloud drizzle icon needs pro plan, so the cloud rain icon is used instead
+    */
+    const get_WeatherIcon = (rangeId) => {
+      switch (true) {
+        case rangeId >= 200 && rangeId < 232:
+          setIcon(weatherIcon.Thunderstorm);
+          break;
+        case rangeId >= 300 && rangeId <= 321:
+          setIcon(weatherIcon.Rain);
+          break;
+        case rangeId >= 500 && rangeId <= 521:
+          setIcon(weatherIcon.Rain);
+          break;
+        case rangeId >= 600 && rangeId <= 622:
+          setIcon(weatherIcon.Snow);
+          break;
+        case rangeId >= 701 && rangeId <= 781:
+          setIcon(weatherIcon.Atmosphere);
+          break;
+        case rangeId === 800:
+          setIcon(weatherIcon.Clear);
+          break;
+        case rangeId >= 801 && rangeId <= 804:
+          setIcon(weatherIcon.Clouds);
+          break;
+        default:
+          setIcon(weatherIcon.Clouds);
+      }
+    }
+    
+  
+    const toCelsius = (temp) => {
+      let cell = Math.floor(temp - 273.15);
+      return cell;
+    }
+
+   useEffect(() =>{
+        fetch( `http://api.openweathermap.org/data/2.5/weather?q=Auckland&appid=${API_KEY}` )
+            .then(res => res.json())
+            .then(response => {
+                    setMain(response.weather[0].main);
+                    setCelsius(toCelsius(response.main.temp));
+                    setMaxTemp(toCelsius(response.main.temp_max));
+                    setMinTemp(toCelsius(response.main.temp_min));
+                    setDescription(response.weather[0].description);
+                    get_WeatherIcon(response.weather[0].id);
+                  });
+
+    },[]);
+    
+    return (
+        <div className="container">
+            
+                <div className="weather-text-box">
+                    <div className="weather-text">Weather</div>
+                </div>
+
+                <div className="current-weather-box">
+            
+                    <div className="icon-container">
+                        {icon}
+                    </div>
+
+                    <div className="discription">
+                        <div className="weather-NZ">{description}|Auckland, NZ</div>
+                        <div className="temp">{celsius}°C</div>
+                        <div className = "temp-bounds">L:{temp_min}°C|H:{temp_max}°C</div>
+                    </div>
+                   
+                </div>
+        </div>
+        
+     )
+    
+  }
+
+
+export default Weather;
+
+// If you want to start measuring performance in your app, pass a function
```

Comment:
It's best to avoid committing code that is commented out as it isn't providing any functionality and may confuse people who may wonder why this has been left behind. If there's important information here that needs to be referred to for documentation purposes, the wiki is the best place for it :)